app/renderer/components/Session/CapabilityControl.js

```diff
@@ -7,24 +7,24 @@ import { INPUT } from '../AntdTypes';
 
 const {dialog} = remote;
 
-
 export default class NewSessionForm extends Component {
 
-  getLocalFilePath (success) {
-    dialog.showOpenDialog((filepath) => {
-      if (filepath) {
-        success(filepath);
+  async getLocalFilePath (success) {
+    await dialog.showOpenDialog({
+      properties: ['openFile'],
+    }).then((files) => {
```

Comment:
I'm not quite sure how it works, but according to [this](https://www.electronjs.org/docs/api/dialog) tutorial the returned value is an object with canceled, filePaths and bookmarks properties

Reply:
I would try to rewrite this method as:

```
let canceled;
let filePaths;
try {
  ({canceled, filePaths} = await dialog.showOpenDialog(mainWindow, {
    properties: ['openFile']
  }));
} catch (e) {
  log.error(e);
}
if (!canceled && !_.isEmpty(filePaths)) {
  success(filePaths);
}
```